src/language-server/builtins.ts

```diff
@@ -251,6 +256,188 @@ export function registerDefaultBuiltins(program: WhackoProgram): void {
     },
   );
 
+  addBuiltinToProgram(
+    program,
+    "types.ref",
+    ({
+      program,
+      module,
+      typeParameters,
+      node,
+      args,
+      caller,
+      getCurrentBlock,
+    }) => {
+      const [inputType, outputType] = typeParameters;
+      const [theParameter] = args;
+
+      const inputIsUsize = typesEqual(
+        inputType,
+        getIntegerType(IntegerKind.USize),
+      );
+
+      if (!isReferenceType(outputType)) {
+        // TODO: Report error diagnostic
+        reportErrorDiagnostic(
+          program,
+          module,
+          "type",
+          node,
+          `The output type of types.ref() should return a reference type.`,
+        );
+        return theInvalidValue;
+      }
+
+      if (!isRawPointerType(inputType) && !inputIsUsize) {
+        // TODO: Report error diagnostic
+        reportErrorDiagnostic(
+          program,
+          module,
+          "type",
+          node,
+          `The input typeof types.ref() should be a RawPointer or a usize.`,
+        );
+        return theInvalidValue;
+      }
+
+      const currentBlock = getCurrentBlock();
+
+      if (inputIsUsize) {
+        const runtimeValue = ensureRuntime(caller, currentBlock, theParameter);
+        const intToPtrInstruction = buildIntToPtrInstruction(
+          caller,
+          currentBlock,
+          outputType as ClassType,
+          runtimeValue,
+        );
+        return createRuntimeValue(intToPtrInstruction, outputType);
+      } else if (isRawPointerType(inputType)) {
+        const value = ensureRuntime(caller, currentBlock, theParameter);
+        return {
+          instruction: value.instruction,
+          kind: ValueKind.Runtime,
+          type: outputType,
+          valueRef: value.valueRef,
+        } as RuntimeValue;
```

Comment:
I think we should be able to shorten this, now that I think about it...
```suggestion
        return createRuntimeValue(value.instruction, outputType);
```

Reply:
No because we need to change the type here. Returning. Anew runtime value with a different type is the right call here.

Reply:
nvm that looks right